### PR TITLE
feat(authorizenet): add authorizenet testing module to test facade usage

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/authorizenet/src/facades/authorize-net-facade.interface.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Action } from '@ngrx/store';
+import { Observable } from 'rxjs';
+
+import { DaffAuthorizeNetTokenResponse } from '../models/response/authorize-net-token-response';
+
+export interface DaffAuthorizeNetFacadeInterface<T extends DaffAuthorizeNetTokenResponse = DaffAuthorizeNetTokenResponse> {
```
Comment (griest024) [inline libs/authorizenet/src/facades/authorize-net-facade.interface.ts:6]: This should extend from `DaffStoreFacade<Action>`

File: libs/authorizenet/src/facades/authorize-net.facade.ts [hunk 2]
```diff
@@ -8,11 +8,12 @@ import { DaffAuthorizeNetModule } from '../authorize-net.module';
 import { selectToken, selectError, selectTokenResponse } from '../selectors/authorize-net.selector';
 import { DaffAuthorizeNetReducersState } from '../reducers/authorize-net-reducers.interface';
 import { DaffAuthorizeNetTokenResponse } from '../models/response/authorize-net-token-response';
+import { DaffAuthorizeNetFacadeInterface } from './authorize-net-facade.interface';
 
 @Injectable({
   providedIn: DaffAuthorizeNetModule
 })
-export class DaffAuthorizeNetFacade<T extends DaffAuthorizeNetTokenResponse> implements DaffStoreFacade<Action> {
+export class DaffAuthorizeNetFacade<T extends DaffAuthorizeNetTokenResponse> implements DaffStoreFacade<Action>, DaffAuthorizeNetFacadeInterface<T> {
```
Comment (griest024) [inline libs/authorizenet/src/facades/authorize-net.facade.ts:16]: This should not implement `DaffStoreFacade<Action>` directly, that should be through the defined interface.

File: libs/authorizenet/src/facades/authorize-net-facade.interface.ts (hunk 1 above)
Comment (lderrickable) [reply libs/authorizenet/src/facades/authorize-net-facade.interface.ts]: done

File: libs/authorizenet/src/facades/authorize-net.facade.ts (hunk 2 above)
Comment (lderrickable) [reply libs/authorizenet/src/facades/authorize-net.facade.ts]: done

**Comment by lderrickable** [comment]
> I think you need to add `@daffodil/authorizenet/testing` to `tsconfig.json`.

yea, you right. added

**Pull request merged**